internal/self_metrics/metrics.go

```diff
@@ -0,0 +1,135 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package self_metrics
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	mexporter "github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/metric"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"go.opentelemetry.io/contrib/detectors/gcp"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/sdk/metric/controller/basic"
+	"go.opentelemetry.io/otel/sdk/metric/sdkapi"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+var (
+	agentMetricsPrefixFormat = "agent.googleapis.com/%s"
+	formatter                = func(d *sdkapi.Descriptor) string {
+		return fmt.Sprintf(agentMetricsPrefixFormat, d.Name())
+	}
+)
+
+type enabledReceivers struct {
+	metric map[string]int
+	log    map[string]int
+}
+
+func GetEnabledReceivers(uc *confgenerator.UnifiedConfig) (enabledReceivers, error) {
+	eR := enabledReceivers{
+		metric: make(map[string]int),
+		log:    make(map[string]int),
+	}
+
+	// Logging Pipelines
+	for _, p := range uc.Logging.Service.Pipelines {
+		for _, rID := range p.ReceiverIDs {
+			rType := uc.Logging.Receivers[rID].Type()
+			eR.log[rType] += 1
+		}
+	}
+
+	// Metrics Pipelines
+	for _, p := range uc.Metrics.Service.Pipelines {
+		for _, rID := range p.ReceiverIDs {
+			rType := uc.Metrics.Receivers[rID].Type()
+			eR.metric[rType] += 1
+		}
+	}
+
+	return eR, nil
+}
+
+func CollectOpsAgentSelfMetrics(uc *confgenerator.UnifiedConfig, death chan bool) error {
+	eR, err := GetEnabledReceivers(uc)
+	if err != nil {
+		return err
+	}
+	log.Println("Found Enabled Receivers : ", eR)
+
+	// Detect GCP credentials
+	ctx := context.Background()
+	res, err := resource.New(ctx,
+		resource.WithDetectors(gcp.NewDetector()),
+		resource.WithTelemetrySDK(),
+	)
+
+	resOpt := basic.WithResource(res)
+
+	// Exporter options
+	opts := []mexporter.Option{
+		mexporter.WithMetricDescriptorTypeFormatter(formatter),
+		mexporter.WithInterval(time.Minute),
+	}
+
+	// Create exporter pipeline
+	pusher, err := mexporter.InstallNewPipeline(opts, resOpt)
+	if err != nil {
+		return fmt.Errorf("Failed to establish pipeline: %v", err)
+	}
+	defer pusher.Stop(ctx)
+
+	// Start meter
+	meter := pusher.Meter("ops_agent/self_metrics")
+
+	// Collect GAUGE metric
+	gaugeObserver, err := meter.AsyncInt64().Gauge("agent/ops_agent/enabled_receivers")
+	if err != nil {
+		return fmt.Errorf("failed to initialize instrument: %v", err)
+	}
+	_ = meter.RegisterCallback([]instrument.Asynchronous{gaugeObserver}, func(ctx context.Context) {
+
+		for key, value := range eR.metric {
+			labels := []attribute.KeyValue{
+				attribute.String("telemetry_type", "metrics"),
+				attribute.String("receiver_type", key),
+			}
+			gaugeObserver.Observe(ctx, int64(value), labels...)
```

Comment:
Maybe too early to say it at this draft stage, but for e2e testing, we'd like to have below scenario covered:

- start agent with config of x receivers enabled
- verify the enabled_receivers metrics are as expected, # of receiver, label value match..etc
- then restart agent with config of y receivers enabled
- verify the enabled_receivers metrics are updated accordingly